### PR TITLE
Add Operator factory

### PR DESCRIPTION
**Pull request by mbartling**
Abstracted away the giant switch case hell. Code is about the same size, but now it should be easier to add an optimizer.

### CODE REVIEW

**Comment by dboyliao** [comment]
Looking good.
I'll test it before I merge it.
I'll let you know I'm done ASAP 👍

File: utensor_cgen/core.py [hunk 1]
```diff
@@ -40,6 +41,8 @@ def generate(self, src_fname: str):
     container.template_vars["placeholders"] = []
     container.add_header('"{}"'.format(header_fname))
 
+    opFactory = OperatorFactory()
+
     print("Parsing {}".format(self.pb_file))
     graph_info, layers = parse_pb(self.pb_file)
```
Comment (dboyliao) [inline utensor_cgen/core.py]: Why don't you refactor these switch cases, Placeholder and Const, altogether with you `OperatorFactory`?

File: utensor_cgen/operators.py [hunk 2]
```diff
@@ -0,0 +1,125 @@
+from .snippets import *  # pylint: disable=W0401
+from .snippets import register_template
+from ._snippets_base import SnippetContainer, Snippet
```
Comment (dboyliao) [inline utensor_cgen/operators.py]: I think this import is irrelevant here.

File: utensor_cgen/operators.py [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+
+
+class _Operator():
+  # Base Class
+  snippet = None
+  name = ""
+
+  def getSnippet(self):
+    return self.snippet
```
Comment (dboyliao) [inline utensor_cgen/operators.py]: Are you trying to make `self.snippet` read only so you write a getter here?
Why not use `@property`?

File: utensor_cgen/core.py (hunk 1 above)
Comment (mbartling) [reply utensor_cgen/core.py]: Short answer, I didn't have much time to work on this. They depend on more than just `op_info` so I just added `**kwargs` to hand this in the future.

File: utensor_cgen/operators.py (hunk 2 above)
Comment (mbartling) [reply utensor_cgen/operators.py]: all 3? Don't I need the first?

File: utensor_cgen/operators.py (hunk 3 above)
Comment (mbartling) [reply utensor_cgen/operators.py]: not to familiar with `@property`, XD If it works better than we can change it :)

Comment (mbartling) [reply utensor_cgen/operators.py]: yeah, `property` seems like the way to go.

File: utensor_cgen/core.py (hunk 1 above)
Comment (mbartling) [reply utensor_cgen/core.py]: Before I move `Placeholder`, this logic needs to change 
```
container.template_vars["placeholders"].append(out_tname)
header_snippet.template_vars["placeholders"].append(out_tname)
```

Alternatively, I can just replace snippet with the `out_tname`.

Comment (dboyliao) [reply utensor_cgen/core.py]: I see.
I think it's reasonable to do it for now.
OK, I'll merge it after I test it all.
I haven't come up with a good way to automatic test it with a develop board, so I have to do it manually.
It'll take some time

Comment (mbartling) [reply utensor_cgen/core.py]: Yeah auto testing these things is a nightmare. Ideally this code shouldn't be mbed specific so it should just run in an arbitrary program.

**Pull request merged**